src/transformStateWithClones.js

```diff
@@ -1,13 +1,58 @@
 'use strict';
-
 /**
  * @param {Object} state
  * @param {Object[]} actions
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const report = [];
+
+  for (const action of actions) {
+    const stateClone =
+      report.length === 0 ? { ...state } : { ...report[report.length - 1] };
+
+    const actionResult = () => {
+      switch (action.type) {
+        case 'addProperties':
+          return addProperties(stateClone, action.extraData);
```

Comment:
Ensure that `action.extraData` is always an object. If not, this could lead to unexpected results when spreading `action.extraData` into `stateClone`.